src/index.rs

```diff
@@ -523,14 +545,18 @@ mod tests {
             kind: Kind::File,
             addrs: vec![],
             target: None,
+            umode: Default::default(),
+            owner: Default::default(),
         }];
         let index_json = serde_json::to_string(&entries).unwrap();
         println!("{}", index_json);
         assert_eq!(
             index_json,
             "[{\"apath\":\"/a/b\",\
              \"kind\":\"File\",\
-             \"mtime\":1461736377}]"
+             \"mtime\":1461736377,\
+             \"umode\":{\"mode\":33277},\
+             \"owner\":{\"user\":null,\"group\":null}}]"
```

Comment:
I also wonder if these should be flattened into the index entry, https://serde.rs/attr-flatten.html.

Reply:
Also, could we omit them if they're null? It seems a bit wasteful to serialize.

Reply:
Hmm. There seem to be a few options of how to fix this. 
How likely do you think it would be to have the owning user defined but not the group or vice versa? If we store an Option<Owner> rather than an Owner { Option, Option } that would solve it, but then either both user and group need to be defined or neither.
Or I could just impl the serialize and deserialize functions manually so it's more compact if null, but that would make the archive difficult to read using other tools - I think it's best to stick with standard serde serialization.
Then there's flattening, and I imagine serde has a way to simply not include null valuesin the serialization - I think flattening is probably the best way to go.

Reply:
I decided to set flatten the Owner struct and `skip_serializing_if` both the user and group are None, so the null values won't get stored unless only one of the user or group are defined, which seems unlikely.

Reply:
I'm not really sure what would be helpful on Windows, but from memory it doesn't have a strongly distinguished file group the way Unix does. So we might want to store only the owner there...

Reply:
I don't really have a Windows development environment at the moment so I'm not going to be able to test and develop Windows functionality for a while.